### DISCUSSION TITLE
BoundaryConditions::constant_exterior(source, 1.0f),  how to expand the image by top-bottom-left-right directions. but i debug it and find its difference, it is right?

### DISCUSSION THREAD

**Discussion by halcomcn** [halide/Halide#8210] [Q&A]
`limit = BoundaryConditions::constant_exterior(source, 1.0f);`

![pic](https://github.com/halide/Halide/assets/29253107/7eb240e3-c5c5-4b30-b1d0-4abc6a54c423)

**Comment by halcomcn** [comment]
according to that picture, i hope it will get [1] result,  but i use the code"limit = BoundaryConditions::constant_exterior(source, 1.0f)",  it print [2] result.

**Reply by halcomcn** [reply]
https://github.com/halide/Halide/discussions/7993

**Reply by halcomcn** [reply]
https://github.com/halide/Halide/discussions/8208

**Reply by halcomcn** [reply]
https://github.com/halide/Halide/discussions/4851

**Comment by abadams** [comment]
This was answered in #8208

**Reply by halcomcn** [reply]
thanks for your reply.  

if a RDom r.x range (-1 , 3),  when limit(x +r.x) is running,( limit = BoundaryConditions::constant_exterior(source, 1.0f); ),  you can see the index: x+r.x = -1(x=0),  we will get a limit(-1) value?     if we can get limit(-1) value, and then limit(-1) will be 1.0f?
